File: leet_code/trees/sum_root_to_leaf.js
const sumRootToLeaf = (root) => {
  if (!root) return 0;
  let sum = 0;
  const traverse = (node, strNum = "") => {
    strNum += node.val;
    if (!node.left && !node.right) sum += parseInt(strNum);
    if (node.left) traverse(node.left, strNum);
    if (node.right) traverse(node.right, strNum);
  };
  traverse(root);
  return sum;
};

File: leet_code/trees/sum_root_to_leaf.rb
def sum_root_to_leaf(root)
  return 0 if !root
  sum = 0
  traverse = lambda do |node, str_num = ""|
    str_num += node.val.to_s
    sum += str_num.to_i if !node.left && !node.right
    traverse.call(node.left, str_num) if node.left
    traverse.call(node.right, str_num) if node.right
  end
  traverse.call(root)
  sum
end
